### ISSUE TITLE
Non-macro function for string-formatting values with a specifier.

### ISSUE THREAD

**Issue by nervecenter** [nim-lang/Nim#21300]
### Summary

I am finding a need for a simple `func` for formatting a value with a specifier and returning the resultant string.

### Description

This simple `func`, in functional style, returns a new string which is the transformation of the input string, integer, or float into a format-specified string. It is cleaner to use inside of other templates or with tables and Json objects.
```nim
# defined in std/strformat

func formatSpec(input: string|SomeInteger|SomeFloat, specifier: string): string =
    formatValue(result, input, specifier)
```

### Alternatives

The `fmt` and `&` macros are convenient for random interpolations, but seem to break inside of other templates (Nimja, code filters, etc.) and have poor behavior and semantics when given tables or Json objects with string keys. For example:

```nim
&"{someTable[\"first_key\"][\"second_key\"]:.2f}"
```
is not very readable and ruins the syntax highlighting algorithm in VSCode's Nim plugin.

### Examples

```nim
echo formatSpec(3.14159, ".2f")
>> 3.14
echo formatSpec(jsonObj["math"]["contants"]["e"].getFloat, ".4f")
>> 2.7183
echo guide["life"]["the_universe"]["everything"].getInt.float.formatSpec("04g")
>> 0042
```

**Comment by Araq** [comment]
Btw instead of `&"{someTable[\"first_key\"][\"second_key\"]:.2f}"` you can use `&"""{someTable["first_key"]["second_key"]:.2f}"""`

Your `formatSpec` needs a better name.

**Comment by nervecenter** [comment]
> Btw instead of `&"{someTable[\"first_key\"][\"second_key\"]:.2f}"` you can use `&"""{someTable["first_key"]["second_key"]:.2f}"""`
> 
> Your `formatSpec` needs a better name.

I agree on the name, it's awful but it communicates the intent. Any improvement on the name is welcome. And thanks for letting me know about the "long string" triple quotes, they're a bit of a footnote in the manual.

I supposed the `formatEng` and `formatFloat` procs in `strutils` get close to what I'm asking for (seems like it'd make more sense for them to be in `strformat`), but I feel my function is a nice-to-have simple alternative.
